### ISSUE TITLE
Kibana 5.3.0 dashboard issue

### ISSUE THREAD

**Issue by met3or** [wazuh/wazuh#94]
Hi guys, 

Firstly I'd like to thank all involved for the great work on this project, it has been really useful for me in some projects! 

I've recently upgraded my Elastic stack from 2.X to 5.X (most recent releases of involved packages, Elasticsearch / Kibana) and the dashboard within this repository doesn't appear to work as expected.

I've modified the wazuh-kibana5-dashboards.json to use my indices, however I get the following error from Kibana:

"Saved "field" parameter is now invalid. Please select a new field.
Visualize: "field" is a required parameter"
"TypeError: "field" is a required parameter"

Upon investigation on some of the visualisations within Edit mode, a variety of the aggregations have their Field empty.

There are no elasticsearch errors produced within the logs for this either.  I've tried re-applying the template, as well as re-indexing the data, ensuring that the wazuh-elastic5-template.json is being applied. 

When searching these errors, the most common advise is to refresh the index patterns, which I have done to no joy.

Any one else experienced this and found a solution?

Thanks!

met3or

**Comment by snaow** [comment]
Hi @met3or,

Thanks for try out Wazuh and thanks for your feedback!

Your current issue is a well-known problem nowadays with Elastic/Kibana, it is not strictly related to Wazuh dashboards.
In general terms, Kibana won't allow anymore to add Visualizations/Dashboards containing fields that do not exist in Elasticsearch indices.

Prior Kibana 5.0.2 version, Dashboards (like Wazuh ones) could contain fields that did not exist yet, meaning that we could import any visualization we wanted, sadly, current Kibana version does not allow us to do that.

One posible solution, as you read, is to refresh the index pattern once you got documents containing all the fields present in the dashboards you want to import (crazy, isn't it?).
Another solution could be use [Wazuh App](https://github.com/wazuh/wazuh-kibana-app), we have implemented a work-around solution for forcing index patterns fields no matter if they exist or not.

Let me also reference here some related issues from Kibana repositories, also including a link to Wazuh community lists where we have discussed this some time ago.

https://groups.google.com/d/msg/wazuh/BC2VzbUc6MI/nIv8OPiPBwAJ
https://github.com/elastic/kibana/issues/9571
https://github.com/elastic/elasticsearch/issues/22438

I hope it helps, let us know if Wazuh App solve the issue.
Regards,
Pedro.

**Comment by met3or** [comment]
Wow, thank you for the super fast response.  Appreciate it.  That all makes sense, as to why the dashboards were working previously!  

I've installed the Wazuh App, but it had complained about the database not existing.  So I had to halt it's progress as I would need to re-install Wazuh I believe.  

Also, I would need to analyse data from multiple sources, not all of which I'm able to access the API directly.

Reading those links, it does look like this issue is perhaps getting some attention and will hopefully be addressed in a future release of ES / Kibana.

Appreciate your input!
